### PR TITLE
GCC: unused warnings

### PR DESCRIPTION
**Pull request by jeffro256**
This PR fixes 4 instances of `-Wunused...`

### CODE REVIEW

File: src/wallet/wallet2.cpp [hunk 1]
```diff
@@ -5127,7 +5124,6 @@ std::string wallet2::exchange_multisig_keys(const epee::wipeable_string &passwor
     expanded_msgs.emplace_back(msg);
 
   // reconstruct multisig account
-  crypto::public_key dummy;
```
Comment (selsta) [inline src/wallet/wallet2.cpp:5130]: This one is already fixed in one of the multisig PRs. Can you undo this change?

Comment (jeffro256) [reply src/wallet/wallet2.cpp]: yup

File: src/wallet/wallet2.cpp [hunk 2]
```diff
@@ -3476,7 +3475,6 @@ void wallet2::refresh(bool trusted_daemon, uint64_t start_height, uint64_t & blo
       if (!first && blocks.empty())
       {
         m_node_rpc_proxy.set_height(m_blockchain.size());
-        refreshed = true;
```
Comment (mj-xmr) [inline src/wallet/wallet2.cpp:3479]: I think this was set by the author to facilitate easier debugging. 

I personally don't need it, but just saying that's the reason I haven't removed it myself.

**Review by mj-xmr — APPROVED** [review]
It looks good, but please gather 2nd or even 3rd opinion about the wallet2.

File: src/crypto/hash-extra-jh.c [hunk 3]
```diff
@@ -38,5 +38,6 @@
 
 void hash_extra_jh(const void *data, size_t length, char *hash) {
   int r = jh_hash(HASH_SIZE * 8, data, 8 * length, (uint8_t*)hash);
+  (void) r; // silence -Wunused-variable
```
Comment (mj-xmr) [inline src/crypto/hash-extra-jh.c:41]: Before anybody mentions inconsistency in the casting style:

This is C, not C++, so `static_cast<>` can't be used.

Comment (hyc) [reply src/crypto/hash-extra-jh.c:41]: So these are only unused because assert() is being no-op'd. These hash functions don't allocate memory so have no reason to fail. Maybe we can just get rid of `r` and the asserts too?

File: src/wallet/wallet2.cpp (hunk 2 above)
Comment (hyc) [reply src/wallet/wallet2.cpp:3479]: That was my thought as well. I don't see any need to remove it.

File: src/crypto/hash-extra-jh.c (hunk 3 above)
Comment (jeffro256) [reply src/crypto/hash-extra-jh.c:41]: Yeah since `hashbitlen` is a compile-time constant, there should be no reason for `jh_hash` to not return `SUCCESS`

File: src/wallet/wallet2.cpp (hunk 2 above)
Comment (jeffro256) [reply src/wallet/wallet2.cpp]: @hyc pushed changes to keep the flag in and remove the warning

File: src/crypto/hash-extra-jh.c (hunk 3 above)
Comment (jeffro256) [reply src/crypto/hash-extra-jh.c]: Okay changes are pushed for that

File: src/wallet/wallet2.cpp (hunk 2 above)
Comment (selsta) [reply src/wallet/wallet2.cpp]: @jeffro256 "Easier debugging" was speculation. As it says in mooo's comment, he wanted to clean it up but probably forgot about it. We can ask him again to be sure but just silencing doesn't make much sense here.

Comment (jeffro256) [reply src/wallet/wallet2.cpp]: Pinging @moneromooo-monero

File: src/crypto/hash-extra-jh.c (hunk 3 above)
Comment (hyc) [reply src/crypto/hash-extra-jh.c]: Hi, sorry for the wishy-washiness but I'm gonna take back what I said and go with your initial patch. Safer to leave the asserts in place so that they're present in Debug builds, and will catch any accidental misuses of the code in the future.

Comment (jeffro256) [reply src/crypto/hash-extra-jh.c]: Should I make the `assert` statements conditional on whether `NDEBUG` is defined?

Comment (hyc) [reply src/crypto/hash-extra-jh.c]: OK, sounds good.

File: src/wallet/wallet2.cpp (hunk 2 above)
Comment (selsta) [reply src/wallet/wallet2.cpp]:
```
12:34 <moneromooo> I had to reorder some code to fix... a timing info leak IIRC. In turn, this undid something I had fixed before, which I believe was a subtle race condition with the txpool.
12:35 <moneromooo> It was pretty subtle IIRC, and so I needed time to think about how to refix it after the move, and I never got to it. I do not remember the details of the race.
12:35 <moneromooo> I don't think I'm likely to ever get to it now.
```

So either we remove the unused variable or we keep it. Silencing doesn't make sense in my opinion.

Comment (mj-xmr) [reply src/wallet/wallet2.cpp]: +1 for commenting out rather than silencing.

There will surely be a lot of debugging to do there in future, but then it can be uncommented on demand, still not causing warnings to everybody else and not having to resort to compiler hacks, which might not be cross-platform.

Comment (hyc) [reply src/wallet/wallet2.cpp]: Probably should add some of mooo's commentary as a comment in the source too. Saves digging thru github issues to find out what it's all about.

Comment (jeffro256) [reply src/wallet/wallet2.cpp:3479]: done

File: src/wallet/wallet2.cpp [hunk 4]
```diff
@@ -3418,7 +3418,10 @@ void wallet2::refresh(bool trusted_daemon, uint64_t start_height, uint64_t & blo
   uint64_t blocks_start_height;
   std::vector<cryptonote::block_complete_entry> blocks;
   std::vector<parsed_block> parsed_blocks;
-  bool refreshed = false;
+  // moneromooo-monero says this about the "refreshed" variable:
+  // "I had to reorder some code to fix... a timing info leak IIRC. In turn, this undid something I had fixed before, ... a subtle race condition with the txpool.
+  // It was pretty subtle IIRC, and so I needed time to think about how to refix it after the move, and I never got to it."
```
Comment (selsta) [inline src/wallet/wallet2.cpp:3423]:
```suggestion
  // TODO: moneromooo-monero says this about the "refreshed" variable:
  // "I had to reorder some code to fix... a timing info leak IIRC. In turn, this undid something I had fixed before, ... a subtle race condition with the txpool.
  // It was pretty subtle IIRC, and so I needed time to think about how to refix it after the move, and I never got to it."
  // https://github.com/monero-project/monero/pull/6097
```

`TODO` so that it can be quickly found via `git grep`. Also adding a URL for context.

Comment (jeffro256) [reply src/wallet/wallet2.cpp]: done!

**Comment by moneromooo-monero** [comment]
At this point I'm unlikely to spend a lot of time thinking about it and fixing it well. So if people want to kill the warning, be my guest. I'd be better if people wanted to fix it properly instead of just killing the warning, but that's up to them.

**Comment by selsta** [comment]
@jeffro256 Can you keep the `refreshed` variable untouched and only add the comment? We can keep the warning as a reminder for someone to look into, if it stays unfixed for another year then we can decide what to do next with it.

Sorry for this huge discussion :)

**Comment by jeffro256** [comment]
@selsta done!

**Review by mj-xmr — APPROVED** [review]
